qa/standalone/misc/ok-to-upgrade.sh

```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/env bash
+#
+# Copyright (C) 2025 IBM
+#
+# Author: Sridhar Seshasayee <sseshasa@redhat.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Library Public License as published by
+# the Free Software Foundation; either version 2, or (at your option)
+# any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Library Public License for more details.
+#
+
+source $CEPH_ROOT/qa/standalone/ceph-helpers.sh
+
+function run() {
+    local dir=$1
+    shift
+
+    export CEPH_MON="127.0.0.1:7150" # git grep '\<7150\>' : there must be only one
+    export CEPH_ARGS
+    CEPH_ARGS+="--fsid=$(uuidgen) --auth-supported=none "
+    export ORIG_CEPH_ARGS="$CEPH_ARGS"
+
+    local funcs=${@:-$(set | ${SED} -n -e 's/^\(TEST_[0-9a-z_]*\) .*/\1/p')}
+    for func in $funcs ; do
+        setup $dir || return 1
+        $func $dir || return 1
+        kill_daemons $dir KILL || return 1
+        teardown $dir || return 1
+    done
+}
+
+function TEST_ok_to_upgrade_invalid_args() {
+    local dir=$1
+    CEPH_ARGS="$ORIG_CEPH_ARGS --mon-host=$CEPH_MON "
+
+    run_mon $dir a --public-addr=$CEPH_MON || return 1
+    run_mgr $dir x || return 1
+    run_osd $dir 0 --osd-mclock-skip-benchmark=true || return 1
+
+    # test with no args
+    ! ceph osd ok-to-upgrade || return 1
+
+    # test with invalid crush bucket name
+    local invalid_bucket="foo"
+    local ceph_version_short="01.00.00-gversion-test"
+    ! ceph osd ok-to-upgrade $invalid_bucket $ceph_version_short || return 1
+}
+
+function TEST_ok_to_upgrade_replicated_pool() {
+    local dir=$1
+    local poolname="test"
+    local OSDS=10
+    local upgrade_version="01.00.00-gversion-test"
+
+    CEPH_ARGS="$ORIG_CEPH_ARGS --mon-host=$CEPH_MON "
+
+    run_mon $dir a --public-addr=$CEPH_MON || return 1
+    run_mgr $dir x || return 1
+
+    for osd in $(seq 0 $(expr $OSDS - 1))
+    do
+      run_osd $dir $osd --osd-mclock-skip-benchmark=true || return 1
+    done
+
+    create_pool $poolname 32 32
+    ceph osd pool set $poolname min_size 1
+    sleep 5
+
+    wait_for_clean || return 1
+
+    # Test for upgradability with min_size=1
+    local exp_osds_upgradable=3
+    local hostname=$(ceph osd tree | grep host | awk '{ print $4 }')
+    local res=$(ceph osd ok-to-upgrade $hostname $upgrade_version --format=json)
```

Comment:
I see you're using $hostname and $upgrade_version, I think if we were to keep the semantics the same through out this PR, you might want to consider `crush_bucket` and `ceph_version` for consistency reasons, but again not a big deal

Reply:
Sure, changed as per your suggestion.